File: microstarter-exchange/src/assets/clients/EndpointClient.js


import axios from 'axios';
let EndpointClient ={

    findAll(fn) {
        axios.get('http://localhost:8787/api/v1/endpoint/findAll'
            // ,
            // {
            //     headers: {
            //         'Authorization': token
            //     }
            // }
        ).then(response => fn(response)).catch(error => console.log(error))
    },
    findByWebService(service,fn){

        axios.get('http://localhost:8787/api/v1/endpoint/getByService?service='+service
        ).then(response => fn(response))
            .catch(error => console.log(error))
    },

    findById(id, fn) {
        axios
            .get('http://localhost:8787/api/v1/endpoint/get?id=' + id
                // ,
                // {
                //     headers: {
                //         'Authorization': token
                //     }
                // }
            )
            .then(response => fn(response))
            .catch(error => console.log(error))
    },

    create(context, fn) {


        console.log("I'm here");
        axios
            .post('http://localhost:8787/api/v1/endpoint/save', context)
            .then(response => fn(response))
            .catch(error => console.log(error))
    },

    update(id, context, fn) {
        axios
            .put('http://localhost:8787/api/v1/endpoint/update', context
                // ,
                // {
                //     headers: {
                //         'Authorization': token
                //     }
                // }
            )
            .then(response => fn(response))
            .catch(error => console.log(error))
    },

    delete(id, fn) {
        axios
            .delete('http://localhost:8787/api/v1/endpoint/delete/' + id
                // ,
                // {
                //     headers: {
                //         'Authorization': token
                //     }
                // }
            )
            .then(response => fn(response))
            .catch(error => console.log(error))
    }
}

export default EndpointClient;

File: MicrostarterExchange/src/main/java/io/hashimati/clients/EndpointClient.java
package io.hashimati.clients;


import io.micronaut.context.annotation.Parameter;
import io.micronaut.http.annotation.*;
import io.micronaut.http.client.annotation.Client;
import jakarta.inject.Inject;
import io.hashimati.domains.Endpoint;

import io.micronaut.http.MediaType;
import io.micronaut.http.multipart.CompletedFileUpload;





@Client("/api/v1/endpoint")
public interface EndpointClient {

    @Post("/save")
    public Endpoint save(Endpoint endpoint, @Header("Authorization") String authorization);

    @Get("/get")
    public Endpoint findById(@Parameter("id") String id, @Header("Authorization") String authorization);

    @Delete("/delete/{id}")
    public boolean deleteById(@PathVariable("id") String id, @Header("Authorization") String authorization);

    @Get("/findAll")
    public Iterable<Endpoint> findAll(  @Header("Authorization") String authorization);

    @Put("/update")
    public Endpoint update(@Body Endpoint endpoint, @Header("Authorization") String authorization);


}


